server/apps/archived/archived.py

```diff
@@ -164,36 +164,44 @@ def on_delete(self, doc):
         doc[config.ID_FIELD] = id_field
 
     def delete(self, lookup):
+        if app.testing and len(lookup) == 0:
+            super().delete(lookup)
+            return
+
+    def on_deleted(self, doc):
+        pass
+
+    def update(self, id, updates, original):
```

Comment:
Because the kill now is divided into 2 steps. 1 - check if kill is allowed on the item and if valid apply the kill template on the item and show it in a modal, allowing the user to modify the reasons for killing the item 2 - when user clicks on the button in the modal the update performs the actual kill process.

Also, the delete method ignores the body in request. Although update isn't the right way because of delete's limitation we had to opt for either patch/post.